src/history.rs

```diff
@@ -158,6 +153,40 @@ impl History {
         wtr.write_all(b"\n")?;
         // https://github.com/rust-lang/rust/issues/32677#issuecomment-204833485
         wtr.flush()?;
+        self.new_entries = 0;
+        Ok(())
+    }
+
+    /// Append new entries in the specified file.
+    // Like [append_history](http://tiswww.case.edu/php/chet/readline/history.html#IDX30).
+    pub fn append<P: AsRef<Path> + ?Sized>(&mut self, path: &P) -> Result<()> {
+        use fs2::FileExt;
+
+        if self.is_empty() || self.new_entries == 0 {
+            return Ok(());
+        }
+        let path = path.as_ref();
+        if !path.exists() || self.new_entries == self.max_len {
+            return self.save(path);
+        }
+        let file = File::open(path)?;
+        file.lock_exclusive()?;
+        // we may need to truncate file before appending new entries
+        let mut other = Self {
+            entries: VecDeque::new(),
+            max_len: self.max_len,
+            ignore_space: self.ignore_space,
+            ignore_dups: self.ignore_dups,
+            new_entries: 0,
+        };
+        other.load(path)?;
```

Comment:
I am not sure we can open `path` in `load` after `file.lock_exclusive`.

Reply:
Ok, it works only on unix (or at least linux) but not on windows (see error in CI)...